### PR TITLE
feat(eslint-plugin): Add `prefer-design-system-colors` rule.

### PR DESCRIPTION
**Pull request by doniyor2109**

### CODE REVIEW

**Comment by umidbekkarimov** [comment]
Let's make it simpler. 
1. Check how this works https://github.com/intellicode/eslint-plugin-react-native/blob/HEAD/docs/rules/no-color-literals.md
2. Do not hardcode dependency to sd/ui, allow to mass an object
```json
{
  "replace": {
    "#fff": { "import": { "source": "@superdispatch/ui", "specifier": "Color" }, "replacer": "Color.White" }
  }
}
```

**Comment by doniyor2109** [comment]
> Let's make it simpler.
> 
> 2. Do not hardcode dependency to sd/ui, allow to pass an object
> 
> ```json
> {
>   "replace": {
>     "#fff": { "import": { "source": "@superdispatch/ui", "specifier": "Color" }, "replacer": "Color.White" }
>   }
> }
> ```

ESLint schema requires all options to be predefined. 

`..."#fff": { "imp...` wont work for us.

This could work for us:

```json

{
  "colors": [
    ["#fff", "White"]
  ],
  "specifier": "Color",
  "source": "@superdispatch/ui"
}
```

**Comment by doniyor2109** [comment]
Oh, there is an option to allow additional properties `additionalProperties`.

**Comment by umidbekkarimov** [comment]
> > Let's make it simpler.
> > 
> > 1. Do not hardcode dependency to sd/ui, allow to pass an object
> > 
> > ```json
> > {
> >   "replace": {
> >     "#fff": { "import": { "source": "@superdispatch/ui", "specifier": "Color" }, "replacer": "Color.White" }
> >   }
> > }
> > ```
> 
> ESLint schema requires all options to be predefined.
> 
> `..."#fff": { "imp...` wont work for us.
> 
> This could work for us:
> 
> ```json
> {
>   "colors": [
>     ["#fff", "White"]
>   ],
>   "specifier": "Color",
>   "source": "@superdispatch/ui"
> }
> ```

How is that different?


> Oh, there is an option to allow additional properties `additionalProperties`.

Well, don't couple rule with sd/ui then.

File: packages/eslint-plugin/rules/__tests__/no-custom-color.spec.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        specifier: 'Color.Grey300',
+      },
+    },
+  },
+});
+
+ruleTester.run('no-custom-color', rule, {
+  valid: [
+    { code: 'const color =  "#e4e7ea";' },
+    { code: 'export const icon = <svg fill="#e4e7ea"></svg>;' },
```
Comment (umidbekkarimov) [inline packages/eslint-plugin/rules/__tests__/no-custom-color.spec.js:29]: so it's actually allows using custom colors :)

Comment (doniyor2109) [reply packages/eslint-plugin/rules/__tests__/no-custom-color.spec.js:29]: Yea) I guess rule name misleading. How about `prefer-colors` ?

Comment (umidbekkarimov) [reply packages/eslint-plugin/rules/__tests__/no-custom-color.spec.js:29]: `prefer-design-system-colors` then

**Pull request merged**